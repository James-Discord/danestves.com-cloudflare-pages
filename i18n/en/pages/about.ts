export const about = {
  title: 'About me',
  paragraphs: [
    'Since I was a child I have been interested in computers starting from video games, every time I played I felt a need to know how it is working and how my character makes any movement',
    'When I got to high school I started researching computers and became interested in how websites are built on top of all the social networks, I liked the idea that when a user selected a menu it showed more options and animations, and that&apos;s what made me fall in love with web development.',
    'I started five years ago to work and learn from the frontend, the visual part of the web; today I consider myself a fullstack because thanks to the knowledge I have gained I can build applications from views to databases and routes. I like to learn more every day, integrate new technologies and contribute to new projects to improve productivity.',
  ],
  buttons: ['Watch CV', 'Download CV'],
}

export default about
