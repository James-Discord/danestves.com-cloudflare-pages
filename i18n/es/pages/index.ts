// Internals
import { posts } from './posts'

export const pages = {
  posts,
}

export default pages
