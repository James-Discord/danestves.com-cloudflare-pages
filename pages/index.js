import { useGraphQL } from 'graphql-react'

export default () => {
  const { loading, cacheValue = {} } = useGraphQL({
    fetchOptionsOverride (options) {
      options.url = 'https://danestves.herokuapp.com/graphql'
    },
    operation: {
      query: /* GraphQL */ `
        {
          blogs {
            title
          }
        }
      `
    }
  })
  console.log(process.env.GITHUB_PERSONAL_ACCESS_TOKEN)

  const { data } = cacheValue
  return data ? (
    <>
      <h1>Hola {data.blogs[0].title}</h1>
      <div style={{ display: 'none' }}>{process.env.GITHUB_PERSONAL_ACCESS_TOKEN}</div>
    </>
  ) : loading ? (
    <p>Loading…</p>
  ) : (
    <p>Error!</p>
  )
}

// import React from 'react'
// import Container from '@material-ui/core/Container'
// import Typography from '@material-ui/core/Typography'
// import Box from '@material-ui/core/Box'
// import MuiLink from '@material-ui/core/Link'
// import { useGraphQL } from 'graphql-react'
// import ProTip from '../src/ProTip'
// import Link from '../src/Link'

// function Copyright () {
//   return (
//     <Typography variant='body2' color='textSecondary' align='center'>
//       {'Copyright © '}
//       <MuiLink color='inherit' href='https://material-ui.com/'>
//         Your Website
//       </MuiLink>{' '}
//       {new Date().getFullYear()}
//       {'. Built with '}
//       <MuiLink color='inherit' href='https://material-ui.com/'>
//         Material-UI.
//       </MuiLink>
//     </Typography>
//   )
// }

// export default function Index () {
//   const { loading, cacheValue: { data, ...errors } = {} } = useGraphQL({
//     fetchOptionsOverride (options) {
//       options.url = 'https://api.github.com/graphql'
//       options.headers = {
//         Authorization: `Bearer ${process.env.GITHUB_PERSONAL_ACCESS_TOKEN}`
//       }
//     },
//     operation: {
//       query: /* GraphQL */ `
//         {
//           viewer {
//             login
//           }
//         }
//       `
//     }
//   })

//   return (
//     <h1>Hola {data && data.viewer.login}</h1>
//   )

//   // return (
//   //   <Container maxWidth='sm'>
//   //     <Box my={4}>
//   //       <Typography variant='h4' component='h1' gutterBottom>
//   //         Next.js example
//   //       </Typography>
//   //       <Link href='/about' color='secondary'>
//   //         Go to the about page
//   //       </Link>
//   //       <ProTip />
//   //       <Copyright />
//   //     </Box>
//   //   </Container>
//   // )
// }
