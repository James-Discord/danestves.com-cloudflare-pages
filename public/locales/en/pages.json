{
  "about": {
    "seo": {
      "title": "About Daniel Esteves ðŸ“‘",
      "description": "I love to code, but I am curious. Every day I try to be better at something and that's why I keep learning, to improve a skill or learn a new one."
    },
    "title": "About me",
    "paragraphs": [
      "Since I was a child I have been interested in computers starting from video games, every time I played I felt a need to know how it is working and how my character makes any movement",
      "When I got to high school I started researching computers and became interested in how websites are built on top of all the social networks, I liked the idea that when a user selected a menu it showed more options and animations, and that&apos;s what made me fall in love with web development.",
      "I started from the age of sixteen to work and learn from the frontend, the visual part of the web; today I consider myself a fullstack because thanks to the knowledge I have acquired I can build applications from views to databases and routes. I like to learn more every day, integrate new technologies and contribute to new projects to improve productivity."
    ],
    "buttons": ["LinkedIn", "Watch CV"]
  },
  "posts": {
    "seo": {
      "title": "The digital garden ðŸŒ± of Daniel Esteves",
      "description": "Read about web development, web applications, React, Vue, JavaScript, TypeScript, Tailwind, and much more on Daniel Esteves' blog."
    }
  }
}
