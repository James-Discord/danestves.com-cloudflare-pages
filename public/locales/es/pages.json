{
  "about": {
    "seo": {
      "title": "Sobre Daniel Esteves üìë",
      "description": "Me encanta programar, pero soy curioso. Cada d√≠a intento ser mejor en algo y por eso sigo aprendiendo, para mejorar una habilidad o aprender una nueva."
    },
    "title": "Sobre m√≠",
    "paragraphs": [
      "Desde peque√±o me he interesado por la inform√°tica empezando por los videojuegos, cada vez que jugaba sent√≠a la necesidad de saber c√≥mo est√° funcionamiento y c√≥mo mi personaje hace cualquier movimiento",
      "Cuando llegu√© al instituto empec√© a investigar sobre inform√°tica y me interes√© por c√≥mo se construyen los sitios web sobre todas las redes sociales, me gustaba la idea de que cuando un usuario seleccionaba un men√∫ mostraba m√°s opciones y animaciones, y eso fue lo que me hizo enamorarme del desarrollo web.",
      "Empec√© desde los diecis√©is a√±os a trabajar y aprender del frontend, la parte visual de la web; hoy me considero un fullstack porque gracias a los conocimientos que he adquirido puedo construir aplicaciones desde vistas hasta bases de datos y rutas. Me gusta aprender cada d√≠a m√°s, integrar nuevas tecnolog√≠as y contribuir a nuevos proyectos para mejorar la productividad."
    ],
    "buttons": ["LinkedIn", "Ver CV"]
  },
  "posts": {
    "seo": {
      "title": "El jard√≠n digital üå± de Daniel Esteves",
      "description": "Lee sobre desarrollo web, aplicaciones web, React, Vue, JavaScript, TypeScript, Tailwind, y mucho m√°s en el blog de Daniel Esteves."
    }
  }
}
