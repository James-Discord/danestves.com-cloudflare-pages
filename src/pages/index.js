import React from "react"
import { Link, graphql } from "gatsby"

import Layout from "../components/layout"
import SEO from "../components/seo"
import {
  Coding,
  Services,
  InterfaceIcon,
  DesignSystemIcon,
} from "../components"

const IndexPage = ({ data }) => (
  <Layout>
    <SEO title="Home" />

    <div className="flex flex-wrap items-center py-12">
      <div className="order-2 w-full md:w-3/5 md:order-1">
        <h1 className="mt-5 text-6xl font-bold leading-none text-indigo-800 text-center md:text-left md:mt-0">
          Daniel Esteves
        </h1>
        <h2 className="text-3xl font-normal text-gray-700 text-center md:text-left">
          Desarrollador web frontend
        </h2>
        <p className="mt-3 text-gray-600 text-1xl text-center md:text-left">
          Desarrollando sitios web utilizando WordPress, React, Gatsby, NextJS y
          mucho más. Listo para realizar tu sueño.
        </p>

        <div className="flex mt-5">
          <Link
            to="/curriculum"
            className="w-1/2 px-8 py-2 mr-2 text-xl font-semibold text-indigo-700 capitalize bg-transparent border border-indigo-700 rounded hover:bg-indigo-700 hover:text-white text-center hover:border-transparent transition-colors transition-300 transition-linear"
          >
            curriculum
          </Link>

          <Link
            to="/contacto"
            className="w-1/2 px-8 py-2 ml-2 text-xl font-semibold text-white capitalize bg-indigo-700 rounded hover:bg-indigo-600 text-center transition-colors transition-300 transition-linear"
          >
            contactame
          </Link>
        </div>
      </div>

      <div className="order-1 w-full md:w-2/5 md:order-2">
        <Coding className="w-64 h-64 mx-auto drop-shadow" />
      </div>
    </div>

    <h2 className="mt-5 mb-12 text-5xl font-medium leading-none text-indigo-800 text-center">
      ¿Cómo haré realidad tu sueño?
    </h2>

    <div className="flex flex-wrap items-center mb-10">
      <div className="order-2 w-full md:w-3/5 md:order-1">
        <h3 className="mb-5 text-4xl font-medium leading-none text-indigo-800">
          Interfaces limpias
        </h3>

        <p className="mb-5 text-2xl text-gray-700 md:mb-0">
          Al momento de realizar los prototipos de tus diseños se harán en base
          a estándares y accesibilidad con el cual tu sitio se verá muy limpio y
          rápido.
        </p>
      </div>
      <div className="order-1 w-full md:w-2/5 md:order-2">
        <DesignSystemIcon className="w-56 h-56 mx-auto mb-5 d-block md:mb-0" />
      </div>

      <div className="order-4 w-full md:w-2/5 md:order-3">
        <InterfaceIcon className="w-56 h-56 mx-auto mb-5 d-block md:mb-0" />
      </div>
      <div className="order-4 w-full md:w-3/5 md:order-3">
        <h3 className="mb-5 text-4xl font-medium leading-none text-indigo-800 text-right">
          Adaptable a todos los dispositivos
        </h3>

        <p className="mb-5 text-2xl text-gray-700 text-right md:mb-0">
          Al momento de realizar los prototipos de tus diseños se harán en base
          a estándares de diseño y accesibilidad con el cual el diseño de tu
          sitio se verá muy limpio y rápido.
        </p>
      </div>
    </div>

    <Services
      services={data.allStrapiSkills.nodes}
      options={{
        type: "carousel",
        controls: true,
        perView: 3,
        focusAt: "center",
        breakpoints: {
          768: {
            perView: 2,
          },
          640: {
            perView: 1,
          },
        },
      }}
    />
  </Layout>
)

export const query = graphql`
  query Services {
    allStrapiSkills {
      nodes {
        icon
        title
        content
      }
    }
  }
`

export default IndexPage
