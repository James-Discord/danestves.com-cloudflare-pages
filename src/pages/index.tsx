// Dependencies
import { GetStaticProps, NextPage } from 'next'
import Image from 'next/image'

// Components
import { Link, BlogCard } from '@/components'

// Interfaces
import { FrontMatterPost } from '@/interfaces'

// Libraries
import { getAllFilesFrontMatter } from '@/lib/mdx'

interface Props {
  posts: FrontMatterPost[]
}

const Index: NextPage<Props> = ({ posts }) => {
  return (
    <>
      <div className="relative flex items-center py-32 lg:py-20">
        <div className="absolute top-0 left-0 w-full h-full bg-secondary">
          <Image
            src="/hero.webp"
            alt="Daniel Esteves - Programador web fullstack en JavaScript"
            layout="fill"
            objectFit="cover"
            objectPosition="center"
          />
          <div className="absolute inset-0 bg-opacity-75 bg-secondary" />
        </div>
        <div className="container z-20 grid items-center grid-cols-12 px-5">
          <div className="col-span-12 md:col-span-5">
            <svg
              viewBox="0 0 1080 1080"
              className="w-48 h-48 mx-auto mb-4 md:mb-0 md:w-56 lg:w-64 md:h-56 lg:h-64 text-primary"
            >
              <path
                d="M1061.237 540.246c-.105 288.558-237.61 520.991-526.204 520.991H207.654A188.891 188.891 0 0118.763 872.346V705.883a23.728 23.728 0 0123.71-23.727h24.166a46.911 46.911 0 0146.894 46.929V872.24a94.209 94.209 0 0094.226 94.226h327.608c235.697 0 430.275-189.278 431.1-424.975C967.292 305.25 776.013 113.533 540 113.533H207.76a94.226 94.226 0 00-94.227 94.191v143.227a46.911 46.911 0 01-46.894 46.893H42.473a23.71 23.71 0 01-23.71-23.692V207.654A188.927 188.927 0 01207.654 18.763H540c287.944 0 521.36 233.416 521.237 521.483z"
                id="prefix__path3982"
                fill="currentColor"
                strokeWidth={1.755}
              />
              <path
                d="M871.696 542.527c-1.368 182.907-152.984 329.17-335.855 329.17H255.232a46.929 46.929 0 01-46.928-46.894V706.04a23.886 23.886 0 0123.868-23.885h46.77a24.114 24.114 0 0124.132 24.149v58.582a12.04 12.04 0 0012.04 12.039h221.885c130.625 0 238.927-104.458 239.927-235.066A236.926 236.926 0 00540 303.074H315.113a12.04 12.04 0 00-12.039 12.04v58.599a24.131 24.131 0 01-24.131 24.131h-46.771a23.903 23.903 0 01-23.868-23.85V255.232a46.946 46.946 0 0146.928-46.928H540c184.065 0 333.1 149.877 331.696 334.223z"
                id="prefix__path3984"
                fill="currentColor"
                strokeWidth={1.755}
              />
              <path
                d="M587.385 542.352c-1.228 25.482-23.201 45.033-48.701 45.033H42.332a23.552 23.552 0 01-23.57-23.534v-47.649a23.552 23.552 0 0123.57-23.587h497.651a47.385 47.385 0 0147.402 49.737z"
                id="prefix__path3986"
                fill="currentColor"
                strokeWidth={1.755}
              />
            </svg>
          </div>
          <div className="col-span-12 md:col-span-7">
            <h1 className="mb-2 text-6xl font-bold leading-none text-center md:text-left text-primary">
              Daniel Esteves
            </h1>
            <h2 className="font-mono text-xl text-center md:text-left text-primary">
              Desarrollador Web Frontend
            </h2>
          </div>
        </div>
      </div>

      <div className="w-full bg-white">
        <div className="container relative z-10 px-5 pb-32 -mt-24">
          <div className="z-30 w-48 h-48 mx-auto border-white rounded-full border-10">
            <Image
              src="/me.jpg"
              alt="Daniel Esteves"
              width={192}
              height={192}
              className="rounded-full"
            />
          </div>

          <h2 className="mt-4 font-mono text-4xl text-center">{`<Hola Mundo />`}</h2>

          <p className="max-w-4xl px-5 mx-auto mt-4 font-mono text-xl text-center">
            Siempre he estado interesado en cómo funciona la tecnología y todo lo que la conforma.
            Desde cómo los personajes se mueven dentro de los videojuegos, hasta el funcionamiento
            de las redes sociales y los sitios web.
          </p>

          <p className="max-w-4xl px-5 mx-auto mt-8 font-mono text-xl text-center">
            Es por eso que decidí convertirme en programador fullstack. Capaz de construir
            aplicaciones, desde la parte visual hasta las bases de datos.
          </p>

          <div className="flex justify-center max-w-4xl mx-auto mt-5">
            <Link
              href="/sobre-mi"
              className="block px-5 py-2 rounded-full bg-secondary text-primary"
            >
              ¡Conóceme más!
            </Link>
          </div>
        </div>
      </div>

      <div className="w-full py-12">
        <div className="container max-w-screen-xl px-5">
          <h2 className="text-3xl font-bold text-center text-primary">Últimos posts:</h2>

          <div className="gap-6 mt-24 md:grid md:grid-cols-2 lg:grid-cols-3">
            {posts.map((post) => (
              <BlogCard key={post.slug} {...post} />
            ))}
          </div>
        </div>
      </div>

      <div className="w-full py-12 bg-primary">
        <div className="max-w-4xl px-5 mx-auto">
          <h2 className="text-3xl font-bold text-center text-secondary">
            Manejo de tecnologías como:
          </h2>

          <div className="grid grid-cols-2 gap-8 mt-8 md:grid-cols-4">
            <div className="text-center">
              <div className="mx-auto">
                <Image src="/static/react.svg" width={128} height={128} alt="React logo" />
              </div>
              <h2 className="font-mono text-xl font-bold text-secondary">
                React / NextJS / Gatsby
              </h2>
            </div>
            <div className="text-center">
              <div className="mx-auto">
                <Image src="/static/nodejs.svg" width={128} height={128} alt="NodeJS logo" />
              </div>
              <h2 className="font-mono text-xl font-bold text-secondary">NodeJS</h2>
            </div>
            <div className="text-center">
              <div className="mx-auto">
                <Image src="/static/graphql.svg" width={128} height={128} alt="GraphQL logo" />
              </div>
              <h2 className="font-mono text-xl font-bold text-secondary">GraphQL</h2>
            </div>
            <div className="text-center">
              <div className="mx-auto">
                <Image src="/static/wordpress.svg" width={128} height={128} alt="WordPress logo" />
              </div>
              <h2 className="font-mono text-xl font-bold text-secondary">WordPress</h2>
            </div>
          </div>
        </div>
      </div>
    </>
  )
}

export const getStaticProps: GetStaticProps = async () => {
  const posts = await getAllFilesFrontMatter('blog')
  const sortedPosts = posts.sort((a: FrontMatterPost, b: FrontMatterPost) => {
    return new Date(b.publishedAt).getTime() - new Date(a.publishedAt).getTime()
  })

  return {
    props: {
      posts: sortedPosts.slice(0, 3),
    },
  }
}

export default Index
