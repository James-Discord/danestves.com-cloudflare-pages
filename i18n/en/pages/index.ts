// Internals
import { about } from './about'
import { notFound } from './notFound'
import { posts } from './posts'

export const pages = {
  about,
  notFound,
  posts,
}

export default pages
