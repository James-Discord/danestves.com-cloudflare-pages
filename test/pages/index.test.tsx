// Dependencies
import * as React from 'react'

// Internals
import { HomePage } from '@/pages'
import { render } from '../testUtils'
import getVideos from '../__mocks__/youtube/videos'

describe('HomePage', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <HomePage
        videos={{
          videos: getVideos(),
        }}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
