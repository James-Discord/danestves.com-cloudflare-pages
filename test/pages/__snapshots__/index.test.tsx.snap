// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot 1`] = `
<DocumentFragment>
  <section
    __source="[object Object]"
    class="container relative overflow-hidden"
    id="danestves-section-hero"
  >
    <div
      __source="[object Object]"
      class="absolute mt-2 -translate-x-1/2 -translate-y-1/2 top-1/2 left-1/2 md:mt-3"
    />
    <div
      __source="[object Object]"
      class="w-full max-w-[332px] -mt-16 mx-auto pl-8 pr-5 pt-3 pb-4 bg-opacity-[0.65] bg-primary backdrop-blur-sm rounded-[20px] md:-mt-32 lg:absolute lg:mr-32 lg:mt-auto lg:-translate-y-1/2 lg:top-1/2 lg:right-1/2"
    >
      <h1
        __source="[object Object]"
        class="text-2xl font-black text-white uppercase"
      >
        daniel esteves 
        <span
          __source="[object Object]"
          aria-label="victory hand"
          role="img"
        >
          ✌️
        </span>
      </h1>
      <p
        __source="[object Object]"
        class="mt-1 text-xs text-white"
      >
        Lead Frontend Developer @ 
        <a
          __source="[object Object]"
          href="https://seeed.us"
          rel="noopener noreferrer"
          target="_blank"
        >
          Seeed
        </a>
        . 
        <br
          __source="[object Object]"
        />
         
        <b
          __source="[object Object]"
        >
          Apasionado por la tecnología
        </b>
         con muchas ganas de aprender y esparcir conocimiento por toda LATAM
      </p>
    </div>
    <div
      __source="[object Object]"
      class="hidden absolute bottom-1/4 lg:block lg:right-[13%] xl:right-[20%]"
    />
  </section>
</DocumentFragment>
`;
